docs-src/docs/guides/walkthrough/intro.md

```diff
@@ -0,0 +1,24 @@
+# An Iterative Walkthrough
+
+This guide aims to take you through three increasingly-complex deployment examples using Layer0.
+Successive sections build upon the previous ones, and each deployment can be completed either through the Layer0 CLI directly, or through Terraform using our custom [Layer0 Terraform Provider](/reference/terraform-plugin).
```

Comment:
I think we just need something that serves as a proper introduction to `terraform`.. why do we use it? What is it good for? What does it compare to? Etc... 

Such a file could exist in the `references` section and be linked from somewhere in the introduction. And yes, we do cover some of this in `reference` already, but I feel like the existing `/reference/terraform-plugin` section is more about our Layer0 provider than it is about terraform specifically.